rclcpp/src/rclcpp/node_interfaces/node_clock.cpp

```diff
@@ -24,13 +24,14 @@ NodeClock::NodeClock(
   rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics,
   rclcpp::node_interfaces::NodeGraphInterface::SharedPtr node_graph,
   rclcpp::node_interfaces::NodeServicesInterface::SharedPtr node_services,
-  rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging)
+  rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging,
+  rcl_clock_type_t clock_type)
 : node_base_(node_base),
   node_topics_(node_topics),
   node_graph_(node_graph),
   node_services_(node_services),
   node_logging_(node_logging),
-  ros_clock_(std::make_shared<rclcpp::Clock>(RCL_ROS_TIME))
+  ros_clock_(std::make_shared<rclcpp::Clock>(clock_type))
```

Comment:
would it be nice to change the name into `node_clock_` or something indicates that belongs to node? i am not sure why this is statically set with `RCL_ROS_TIME` though.

Reply:
+1 on changing name as you suggested.

IMO that was just an oversight that no one tried to fix..
The ROS time is non monotonic, so it's a very dangerous default if your code expects time to be linear.

Reply:
> The ROS time is non monotonic, so it's a very dangerous default if your code expects time to be linear.

I mean, yes, but if you don't deal with it then your code has no chance at all to work with simulation.

It's not exactly clear to me what this PR is doing, but I think we absolutely should keep the default as RCL_ROS_TIME, otherwise simulation will just not work properly.  Further, since this PR is changing a public API (the NodeClock constructor), we should absolutely tick-tock the API change (assuming others disagree with me and think we should always require users to pass the clock type in).

Reply:
This PR adds the possibility to select the clock type of a node via rclcpp::Node Options.

Otherwise APIs such as Node::now()  can't be configured to use mononotic time.

Note that this does not change the default (which remains ROS time).

Reply:
> would it be nice to change the name into `node_clock_` or something indicates that belongs to node? i am not sure why this is statically set with `RCL_ROS_TIME` though.
I change the name to `clock_` since the class itself is named `NodeClock`.